### DISCUSSION TITLE
Missing complete API reference

### DISCUSSION THREAD

**Discussion by forman** [pmndrs/jotai#2062] [General]
First of all, thanks for this wonderful library!

As a jotai beginner I'm missing a reference of all the public API offered by the library. For example, the following are only linked from within guides and are hard to recover:

- https://jotai.org/docs/utilities/split
- https://jotai.org/docs/utilities/select
- https://jotai.org/docs/utilities/callback

I can only search for them if I know what to search for. Also, while browsing through the utilities section in the docs, I find individual functions well described in terms of what they do, i.e., their inputs, outputs and mode of operation, but I'm often lacking a description of the _actual purpose_, a description when to use it or what problem it solves.

In addition, I found the following guides are missing in the left-side navigation menu:

- https://jotai.org/docs/guides/vite  (only babel here, would like to see SWC too)
- https://jotai.org/docs/guides/async

**Comment by dai-shi** [comment]
So, it's not missing but hard to discover?
There were similar suggestions before. @sandren Do you have any idea to mitigate it?

**Reply by forman** [reply]
> So, it's not missing but hard to discover?

That's what I believe. I'd love to see an overview page or just with the API functions grouped by purpose.

**Reply by sandren** [reply]
@dai-shi These docs were previously hidden from both the sidebar and table of content so they could only be found via search. Perhaps it's best for the sidebar to be as comprehensive as search, but still use the introduction doc to highlight the most important docs.

https://github.com/pmndrs/jotai/pull/2063 implements this change.

**Comment by dai-shi** [comment]
> lacking a description of the actual purpose

Can you help improving those?
Let's start one by one. Do  you have any specific doc to start with?

**Reply by forman** [reply]
I'm currently evaluating state management libraries for a new project. My focus is on jotai and zustand. But I'm completely new to both. Have used Redux for long tim and happy to start something new, with less boilerplate code.

**Reply by dai-shi** [reply]
So far, what is the function, for example, that you found you didn't understand its purpose?

**Comment by dai-shi** [comment]
> (only babel here, would like to see SWC too)

I'm not sure if SWC works for vite now. @Thisen Do you have any ideas?

**Reply by forman** [reply]
Hi @dai-shi, thanks for responding that fast :)

I'm using the official [vite-plugin-react-swc](https://github.com/vitejs/vite-plugin-react-swc) since quite a while now.
